### DISCUSSION TITLE
Converting all LINE and ARC into LWPOLYLINE

### DISCUSSION THREAD

**Discussion by PsymonsB** [mozman/ezdxf#695] [Ideas]
Is there a simple way to combine all the data of lines and arcs I know to be continuous into one LWPOLYLINE, currently I'm trying by collecting all the data from arcs and lines to create an array of points and bulges then create a LWPOLYLINE from them, but doing this didn't create the desired effect I now realise the points have to be in the correct continuous order, would it be better for me to organise my points or access the polyline and add a new point in for the new element at its correct position. both seem long winded.
this is needed for a CNC machine that only reads DXF polylines and the output from a certain 3D software package outputs it all as lines and arcs.

**Comment by mozman** [comment]
No, there is no easy way, because this is a trivial task for humans, but not algorithmically.

1. group lines and arcs by connection, all elements of a group have one shared start- and end vertex. 
   - problem: coincident lines, 2 shared vertices, but maybe not in the same order
   - problem: coincident arcs, 2 shared vertices, but maybe a circle of 2 arcs
   - problem: ...
   - hint: arcs may have an extrusion vector of (0, 0, -1) to reverse the angle orientation
2. sort the grouped entities by connected start- and end vertices and build a LWPOYLINE, also many special cases to handle like a full circle as arc, closed loops in groups ...

**Reply by PsymonsB** [reply]
Thanks for your prompt reply, I'm realising its going to be very problematic to do algorithmically the more I try.

Edit -- the more I'm getting close the more edge cases I'm discovering or realising may  need to be allowed for.
